agent/engine/metadataservice/write_metadata.go

```diff
@@ -0,0 +1,88 @@
+package metadataservice
+
+import (
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+//	"github.com/cihub/seelog"
+	//docker "github.com/fsouza/go-dockerclient"
+)
+
+const (
+	taskArnRegexExpr = "(:task/)|(arn:aws:ecs:)|:"
```

Comment:
I'd definitely prefer two `strings.SplitN` calls over a regex here.  We know exactly what format the ARN will be in and can use that.